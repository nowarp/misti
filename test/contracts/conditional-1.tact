fun test(): Int {
    let a: Int = 20;
    if (a > 42) {
        a = 23;
    }
    let b: Int = a + 1;
    return b;
}

