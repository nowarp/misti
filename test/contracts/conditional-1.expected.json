{
  "projectName": "conditional-1",
  "functions": [
    {
      "name": "test",
      "cfg": {
        "nodes": [
          {
            "id": 145,
            "stmtID": 2000,
            "srcEdges": [],
            "dstEdges": [
              147
            ]
          },
          {
            "id": 146,
            "stmtID": 2007,
            "srcEdges": [
              147
            ],
            "dstEdges": [
              149,
              150,
              152
            ]
          },
          {
            "id": 148,
            "stmtID": 2006,
            "srcEdges": [
              149
            ],
            "dstEdges": []
          },
          {
            "id": 151,
            "stmtID": 2012,
            "srcEdges": [
              152
            ],
            "dstEdges": [
              154
            ]
          },
          {
            "id": 153,
            "stmtID": 2014,
            "srcEdges": [
              154
            ],
            "dstEdges": []
          }
        ],
        "edges": [
          {
            "id": 147,
            "src": 145,
            "dst": 146
          },
          {
            "id": 149,
            "src": 146,
            "dst": 148
          },
          {
            "id": 150,
            "src": 146,
            "dst": 2012
          },
          {
            "id": 152,
            "src": 146,
            "dst": 151
          },
          {
            "id": 154,
            "src": 151,
            "dst": 153
          }
        ]
      }
    }
  ],
  "contracts": []
}